src/lib/ColorSelect.svelte

```diff
@@ -50,45 +59,35 @@
 		ctx.putImageData(image, 0, 0)
 	}
 
-	function scale_to_ui(okhsv: number[]) {
-		return [clamp_ui(okhsv[0]), clamp_ui(okhsv[1]), clamp_ui(1 - okhsv[2])]
+	function scale_to_ui(okhsv: Okhsv): Okhsv {
+		return { mode: 'okhsv', h: clamp_ui((okhsv.h ?? 0) / 360), s: clamp_ui(okhsv.s), v: clamp_ui(1 - okhsv.v) }
 	}
 
 	function clamp(x: number) {
 		return x < eps ? eps : x > 1 - eps ? 1 - eps : x
 	}
 
-	function clamp_255(value: number) {
-		return value < 0 ? 0 : value > 255 ? 255 : value
-	}
-
 	function clamp_ui(value: number) {
 		return value < 0 ? 0 : value > 1 ? picker_size : value * picker_size
 	}
 
 	async function update_rgb() {
-		const rgb = okhsv_to_srgb(okhsv[0], okhsv[1], okhsv[2]).map(clamp_255)
-
-		r = rgb[0]
-		g = rgb[1]
-		b = rgb[2]
-
-		dispatch('change', { rgb, okhsv })
+		dispatch('change', { rgb, okhsl, okhsv, oklab, oklch })
```

Comment:
🤔 I’m actually unsure about this part—I think this breaks the `bind:color` directive. Is there a better way to dispatch these events?